controllers/controllers/resource/fetcher.go

```diff
@@ -361,3 +365,22 @@ func MapMachineTemplateToVSphereMachineConfigSpec(vsMachineTemplate *vspherev1.V
 	// TODO: OSFamily, Users
 	return vsSpec, nil
 }
+
+func MapMachineTemplateToVSphereMachineConfigSpecWorkers(vsMachineTemplates []vspherev1.VSphereMachineTemplate) (map[string]anywherev1.VSphereMachineConfig, error) {
+	vsSpec := &anywherev1.VSphereMachineConfig{}
+	vsSpecs := make(map[string]anywherev1.VSphereMachineConfig, len(vsMachineTemplates))
+	for _, vsMachineTemplate := range vsMachineTemplates {
+		vsSpec.Spec.MemoryMiB = int(vsMachineTemplate.Spec.Template.Spec.MemoryMiB)
+		vsSpec.Spec.DiskGiB = int(vsMachineTemplate.Spec.Template.Spec.DiskGiB)
+		vsSpec.Spec.NumCPUs = int(vsMachineTemplate.Spec.Template.Spec.NumCPUs)
+		vsSpec.Spec.Template = vsMachineTemplate.Spec.Template.Spec.Template
+		vsSpec.Spec.ResourcePool = vsMachineTemplate.Spec.Template.Spec.ResourcePool
+		vsSpec.Spec.Datastore = vsMachineTemplate.Spec.Template.Spec.Datastore
+		vsSpec.Spec.Folder = vsMachineTemplate.Spec.Template.Spec.Folder
+		vsSpec.Spec.StoragePolicyName = vsMachineTemplate.Spec.Template.Spec.StoragePolicyName
+		vsSpecs[vsMachineTemplate.Name] = *vsSpec
+	}
+
+	// TODO: OSFamily, Users
```

Comment:
Can we add to the comment mentioning this is because these are immutable? I forget if that's the reason though because those fields would need to be ported over too right?